Docker image doesn't work

Thanks for stopping by to let us know something could be better!

#### Environment details
Docker image

#### Steps to reproduce

  1. Run `docker run --rm --user $UID -v $PWD:/in -v $PWD/proto:/out gcr.io/gapic-images/gapic-generator-typescript:latest`

Looks like gapic-generator is missing a module named 'generator'.

```
gapic-generator-typescript: https://github.com/googleapis/gapic-generator-typescript
Latest commit: 
commit 527b5af5cc9b92900dd5f05a0f485dd07ed4fbdb
Author: Alice <65933803+alicejli@users.noreply.github.com>
Date:   Wed Aug 11 17:59:42 2021 -0700

internal/modules/cjs/loader.js:796
    throw err;
    ^

Error: Cannot find module './generator'
Require stack:
- /usr/local/lib/node_modules/@google-cloud/gapic-generator/build/typescript/src/protoc-plugin.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:793:17)
    at Function.Module._load (internal/modules/cjs/loader.js:686:27)
    at Module.require (internal/modules/cjs/loader.js:848:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/@google-cloud/gapic-generator/build/typescript/src/protoc-plugin.js:18:21)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1043:10) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/usr/local/lib/node_modules/@google-cloud/gapic-generator/build/typescript/src/protoc-plugin.js'
  ]
}
--typescript_gapic_out: protoc-gen-typescript_gapic: Plugin failed with status code 1.
child_process.js:642
    throw err;
    ^

Error: Command failed: protoc --plugin=protoc-gen-typescript_gapic=/usr/local/lib/node_modules/@google-cloud/gapic-generator/build/typescript/src/protoc-plugin.js --typescript_gapic_out=/out -I/in /in/proto/protos/synthesizer.proto -I/protos/api-common-protos-master
    at checkExecSyncError (child_process.js:621:11)
    at Object.execFileSync (child_process.js:639:15)
    at Object.<anonymous> (/usr/local/lib/node_modules/@google-cloud/gapic-generator/build/typescript/src/gapic-generator-typescript.js:136:17)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1043:10)
    at internal/main/run_main_module.js:17:11 {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 18,
  stdout: null,
  stderr: null
}
```

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -12,15 +12,19 @@ RUN npm install -g /root/files/package.tgz
 
 # Download and install protoc
 RUN cd /root/files && \
-    wget https://github.com/protocolbuffers/protobuf/releases/download/v3.12.3/protoc-3.12.3-linux-x86_64.zip
-RUN cd /usr/local && unzip /root/files/protoc-3.12.3-linux-x86_64.zip
+    wget https://github.com/protocolbuffers/protobuf/releases/download/v21.5/protoc-21.5-linux-x86_64.zip
+RUN cd /usr/local && unzip /root/files/protoc-21.5-linux-x86_64.zip
 RUN chmod -R ugo+rX /usr/local
 RUN chmod ugo+rx /usr/local/bin/protoc
 
-# Download a copy of API common protos
+# Download googleapis common protos
 RUN cd /root/files && \
-    wget https://github.com/googleapis/api-common-protos/archive/master.zip
-RUN cd /protos && unzip /root/files/master.zip && chmod -R a+rx /protos
+    wget https://github.com/googleapis/googleapis/archive/refs/heads/master.zip
+RUN cd /root/files && unzip master.zip
+RUN cd /root/files && for dir in google/api google/longrunning google/rpc google/type google/iam/v1; do \
+        mkdir -p /protos/$dir; \
+        cp googleapis-master/$dir/*.proto /protos/$dir/; \
+    done
 
 # Add gapic-config-validator plugin
 COPY --from=gcr.io/gapic-images/gapic-config-validator /usr/local/bin/protoc-gen-gapic-validator /usr/local/bin/protoc-gen-gapic-validator

--- a/docker/publish.sh
+++ b/docker/publish.sh
@@ -21,8 +21,7 @@ SCRIPTDIR=`dirname "$0"`
 cd "$SCRIPTDIR"
 cd ..   # now in the package.json directory
 
-git checkout master
-git pull
+# Publishing from the current branch
 sh docker/build.sh
 sh docker/test.sh
 docker tag gapic-generator-typescript gcr.io/gapic-images/gapic-generator-typescript

--- a/docker/start.sh
+++ b/docker/start.sh
@@ -15,7 +15,7 @@ echo 1>&2
 cd /in
 gapic-generator-typescript \
   --gapic-validator_out=. \
-  --common-proto-path /protos/api-common-protos-master \
+  --common-proto-path /protos \
   -I /in \
   --output-dir /out \
   $* \

--- a/docker/test.sh
+++ b/docker/test.sh
@@ -21,10 +21,13 @@ SCRIPTDIR=`dirname "$0"`
 cd "$SCRIPTDIR"
 cd ..   # now in the package.json directory
 
+export TMPDIR=`mktemp -d`
+
 ### Test script pulling the docker image and use it against showcase proto.
 
 # Docker image tag: gapic-generator-typescript:latest.
-DIR_NAME=$TMPDIR/.showcase-typescript
+DIR_NAME=$TMPDIR/showcase-typescript
+echo "Running Docker container test, TMPDIR: $TMPDIR"
 # Remove test directory if it already exists
 rm -rf $DIR_NAME
 # Create new directory showcase-typescript. 
@@ -41,5 +44,6 @@ npm install  # install dependencies
 npm run fix  # format the code
 npm test     # run unit tests
 
-# Test succeed
-echo 'docker test succeeded! '
+# Test succeeded
+echo 'Docker test succeeded! '
+rm -rf $TMPDIR

--- a/package.json
+++ b/package.json
@@ -17,8 +17,8 @@
     "protoc-gen-typescript_gapic": "build/typescript/src/protoc-plugin.js"
   },
   "files": [
-    "build/src/**/*.js",
-    "build/src/**/*.d.ts",
+    "build/typescript/src/**/*.js",
+    "build/typescript/src/**/*.d.ts",
     "build/protos",
     "build/templates"
   ],

